pages/blog/creating-consistency-announcing-asyncapi-spectral-together.md

```diff
@@ -0,0 +1,206 @@
+
+Back in 2018 or 2019, plenty of code was oriented around OpenAPI, tests, and naming. All were focused solely on OpenAPI,
+meaning rulesets themselves couldn’t be applied easily to other formats (Spectral didn’t support them previously).
+
+With the release of [Spectral 6](https://blog.stoplight.io/api-linting-with-spectral-keeps-getting-better), that transition period wrapped up and Spectral is now more flexible than it’s ever been.
+
+## Steps to Utilizing AsyncAPI & Spectral
+
+Linting AsyncAPI documents is quite similar to linting OpenAPI documents, meaning
+most of the information about Spectral you’ll find on the internet will apply to AsyncAPI. Together with AsyncAPI
+initiative, we [maintain a Spectral ruleset](https://meta.stoplight.io/docs/spectral/ZG9jOjUzNDg-async-api-rules) containing a number of useful rules you can leverage in your workflow to help
+you get started.
+
+**Step One:**
+
+To get started, you need to install Spectral first, which can be done [here](https://meta.stoplight.io/docs/spectral/docs/getting-started/2-installation.md). Note that you need to have [npm](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm) or
+[Yarn](https://yarnpkg.com/getting-started/install) installed, running `npm install -g @stoplight/spectral-cli` or `yarn global add @stoplight/spectral-cli` is sufficient
+to get Spectral.
+
+Since we intend to use the AsyncAPI ruleset, we can also consider installing @stoplight/spectral-rulesets, which
+contains the actual ruleset we’ll use. However, it’s an optional step, as the installation of spectral-cli will already
+install spectral-rulesets for you.
+
+**Step Two:**
+
+Once you’ve got all the required dependencies installed, you can start by creating a simple ruleset. To do so,
+create a file called `.spectral.json`. The following template can be used as a decent baseline:
+
+```json
+{
+  // this makes sure our rules apply only to AsyncAPI documents. 
+  // It might be handy in case you have other specs in the directory you intend to lint.
+  "formats": ["asyncapi2"],
+  // this includes the ruleset linked below
+  // https://meta.stoplight.io/docs/spectral/ZG9jOjUzNDg-async-api-rules
+  // Note, that by default only recommended rules are enabled. 
+  // Some rules listed in the article above may not be a fit for you,
+  // therefore we don’t enable them by default.
+  "extends": "spectral:asyncapi",
+  "rules": {
+    // we can add our own rules here
+  }
+}
+```
+
+**Step Three:**
+
+To better facilitate the actual demands, Spectral allows you to create your own rules. These rules will fill the gaps
+and cover use cases unique to your workflow, style guide. While a more thorough reference regarding custom rules is
+available [here](https://meta.stoplight.io/docs/spectral/ZG9jOjI1MTg5-custom-rulesets#adding-rules), a basic rule could look as follows. The rule we’re about to implement will assert the presence and
```

Comment:
```suggestion
and cover use cases unique to your workflow, and style guide. While a more thorough [reference regarding custom rules is available here](https://meta.stoplight.io/docs/spectral/ZG9jOjI1MTg5-custom-rulesets#adding-rules), a basic rule could look as follows. The rule we’re about to implement will assert the presence and
```
## Ambiguous or incomplete anchor text  
A link is incomplete and/or ambiguous if a user has to read the surrounding copy around the link to get an idea of where it could take them. 

**Example:**
`<a href="url">here</a>`

### WCAG 2.1 success criteria for anchor text: 

> “The [purpose of each link](https://www.w3.org/TR/WCAG21/#dfn-purpose-of-each-link) can be determined from the link text alone or the link text together with its [programmatically determined link context (https://www.w3.org/TR/WCAG21/#dfn-programmatically-determined-link-context), except where the purpose of the link would be [ambiguous to users in general](https://www.w3.org/TR/WCAG21/#dfn-ambiguous-to-users-in-general).”

Think about how an HTML button that only says “here” isn’t informative in isolation; a user reliant on a screenreader is forced to read the surrounding copy and headings to figure out where the link will take you.